### ISSUE TITLE
Rule::unique keeps $idColumn = 'id' even when my primary key is named differently

### ISSUE THREAD

**Issue by Kaishiyoku** [laravel/framework#28023]
- Laravel Version: 5.8.7
- PHP Version: 7.2.4
- Database Driver & Version: MariaDB 10.1.31

### Description:
While using the string-equivalent for uniqueness validation the use of `Rule::unique` doesn't work if the primary key isn't named 'id'.

Exception thrown: 
```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'id' in 'where clause' (SQL: select count(*) as aggregate from `changelogs` where `version_number` = 3.7.0 and `id` <> )
```

Details:

The error occurs at the `ValidateAttributes.php` on line 713 where the $idColumn is being set to 'id' even though my model's primary key is named 'version_number':

Changelog.php
```php
class Changelog extends Model
{
    protected $primaryKey = 'version_number';

    public $incrementing = false;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'version_number', 'published_at', 'features', 'bugs', 'other'
    ];

    /**
     * The attributes that should be hidden for arrays.
     *
     * @var array
     */
    protected $hidden = [
        //
    ];

    /**
     * The attributes that should be mutated to dates.
     *
     * @var array
     */
    protected $dates = [
        'created_at',
        'updated_at',
        'published_at',
    ];

    /**
     * @param string $value
     */
    public function setPublishedAtAttribute($value)
    {
        $this->attributes['published_at'] = Carbon::createFromFormat(__('date.short'), $value);
    }
}
```

### Steps To Reproduce:

Create a simple form and Controller method with a uniqueness validation like so:

```php
public function store(Request $request)
    {
        $validationRules = [
            'version_number' => [
                'required',
                Rule::unique('changelogs'),
            ],
        ];

        $validated = $request->validate($validationRules);

        $changelog = new Changelog($validated);
        $changelog->save();

        flash()->success(__('changelog.admin.create.success'));

        return redirect()->route($this->redirectRoute);
    }
```

Then fill the form and submit it.

An exception is being thrown:

```
Illuminate\Database\QueryException thrown with message "SQLSTATE[42S22]: Column not found: 1054 Unknown column 'id' in 'where clause' (SQL: select count(*) as aggregate from `changelogs` where `version_number` = 3.7.0 and `id` <> )"

Stacktrace:
#88 Illuminate\Database\QueryException in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Connection.php:664
#87 Doctrine\DBAL\Driver\PDOException in C:\xampp\htdocs\notes\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\PDOConnection.php:63
#86 PDOException in C:\xampp\htdocs\notes\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\PDOConnection.php:61
#85 PDO:prepare in C:\xampp\htdocs\notes\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\PDOConnection.php:61
#84 Doctrine\DBAL\Driver\PDOConnection:prepare in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Connection.php:326
#83 Illuminate\Database\Connection:Illuminate\Database\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Connection.php:657
#82 Illuminate\Database\Connection:runQueryCallback in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Connection.php:624
#81 Illuminate\Database\Connection:run in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Connection.php:333
#80 Illuminate\Database\Connection:select in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Query\Builder.php:2100
#79 Illuminate\Database\Query\Builder:runSelect in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Query\Builder.php:2088
#78 Illuminate\Database\Query\Builder:Illuminate\Database\Query\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Query\Builder.php:2574
#77 Illuminate\Database\Query\Builder:onceWithColumns in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Query\Builder.php:2089
#76 Illuminate\Database\Query\Builder:get in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Query\Builder.php:2501
#75 Illuminate\Database\Query\Builder:aggregate in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Database\Query\Builder.php:2429
#74 Illuminate\Database\Query\Builder:count in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Validation\DatabasePresenceVerifier.php:55
#73 Illuminate\Validation\DatabasePresenceVerifier:getCount in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Validation\Concerns\ValidatesAttributes.php:730
#72 Illuminate\Validation\Validator:validateUnique in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Validation\Validator.php:398
#71 Illuminate\Validation\Validator:validateAttribute in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Validation\Validator.php:277
#70 Illuminate\Validation\Validator:passes in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Validation\Validator.php:302
#69 Illuminate\Validation\Validator:fails in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Validation\Validator.php:314
#68 Illuminate\Validation\Validator:validate in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Validation\Factory.php:136
#67 Illuminate\Validation\Factory:validate in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Foundation\Providers\FoundationServiceProvider.php:53
#66 Illuminate\Http\Request:Illuminate\Foundation\Providers\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Support\Traits\Macroable.php:110
#65 call_user_func_array in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Support\Traits\Macroable.php:110
#64 Illuminate\Http\Request:__call in C:\xampp\htdocs\notes\app\Http\Controllers\Admin\ChangelogController.php:49
#63 App\Http\Controllers\Admin\ChangelogController:store in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Controller.php:54
#62 call_user_func_array in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Controller.php:54
#61 Illuminate\Routing\Controller:callAction in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\ControllerDispatcher.php:45
#60 Illuminate\Routing\ControllerDispatcher:dispatch in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Route.php:219
#59 Illuminate\Routing\Route:runController in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Route.php:176
#58 Illuminate\Routing\Route:run in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Router.php:680
#57 Illuminate\Routing\Router:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:30
#56 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\app\Http\Middleware\Administrate.php:48
#55 App\Http\Middleware\Administrate:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#54 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#53 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\app\Http\Middleware\Menus.php:109
#52 App\Http\Middleware\Menus:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#51 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#50 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\app\Http\Middleware\Language.php:61
#49 App\Http\Middleware\Language:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#48 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#47 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Middleware\SubstituteBindings.php:41
#46 Illuminate\Routing\Middleware\SubstituteBindings:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#45 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#44 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Auth\Middleware\Authenticate.php:43
#43 Illuminate\Auth\Middleware\Authenticate:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#42 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#41 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\VerifyCsrfToken.php:75
#40 Illuminate\Foundation\Http\Middleware\VerifyCsrfToken:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#39 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#38 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\View\Middleware\ShareErrorsFromSession.php:49
#37 Illuminate\View\Middleware\ShareErrorsFromSession:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#36 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#35 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Session\Middleware\StartSession.php:56
#34 Illuminate\Session\Middleware\StartSession:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#33 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#32 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse.php:37
#31 Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#30 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#29 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Cookie\Middleware\EncryptCookies.php:66
#28 Illuminate\Cookie\Middleware\EncryptCookies:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#27 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#26 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:104
#25 Illuminate\Pipeline\Pipeline:then in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Router.php:682
#24 Illuminate\Routing\Router:runRouteWithinStack in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Router.php:657
#23 Illuminate\Routing\Router:runRoute in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Router.php:623
#22 Illuminate\Routing\Router:dispatchToRoute in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Router.php:612
#21 Illuminate\Routing\Router:dispatch in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php:176
#20 Illuminate\Foundation\Http\Kernel:Illuminate\Foundation\Http\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:30
#19 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\barryvdh\laravel-debugbar\src\Middleware\InjectDebugbar.php:65
#18 Barryvdh\Debugbar\Middleware\InjectDebugbar:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#17 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#16 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\TransformsRequest.php:21
#15 Illuminate\Foundation\Http\Middleware\TransformsRequest:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#14 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#13 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\TransformsRequest.php:21
#12 Illuminate\Foundation\Http\Middleware\TransformsRequest:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#11 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#10 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\ValidatePostSize.php:27
#9 Illuminate\Foundation\Http\Middleware\ValidatePostSize:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#8 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#7 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode.php:62
#6 Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode:handle in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:163
#5 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:53
#4 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:104
#3 Illuminate\Pipeline\Pipeline:then in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php:151
#2 Illuminate\Foundation\Http\Kernel:sendRequestThroughRouter in C:\xampp\htdocs\notes\vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php:116
#1 Illuminate\Foundation\Http\Kernel:handle in C:\xampp\htdocs\notes\public\index.php:53
#0 require_once in C:\xampp\htdocs\notes\server.php:21
```

**Comment by staudenmeir** [comment]
This has been fixed in Laravel 5.8.8 (#27987).

**Comment by staudenmeir** [comment]
Please close the issue.
